kornia/color/hls.py

```diff
@@ -110,45 +101,32 @@ def hls_to_rgb(image: torch.Tensor) -> torch.Tensor:
         >>> input = torch.rand(2, 3, 4, 5)
         >>> output = hls_to_rgb(input)  # 2x3x4x5
     """
-    if not isinstance(image, torch.Tensor):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(image)}")
+    if not isinstance(image, Tensor):
+        raise TypeError(f"Input type is not a Tensor. Got {type(image)}")
 
     if len(image.shape) < 3 or image.shape[-3] != 3:
         raise ValueError(f"Input size must have a shape of (*, 3, H, W). Got {image.shape}")
 
-    if not torch.jit.is_scripting():
-        # weird way to use globals compiling with JIT even in the code not used by JIT...
-        # __setattr__ can be removed if pytorch version is > 1.6.0 and then use:
-        # hls_to_rgb.HLS2RGB = hls_to_rgb.HLS2RGB.to(image.device)
-        hls_to_rgb.__setattr__('HLS2RGB', hls_to_rgb.HLS2RGB.to(image))  # type: ignore
-        _HLS2RGB: torch.Tensor = hls_to_rgb.HLS2RGB  # type: ignore
-    else:
-        _HLS2RGB = torch.tensor([[[0.0]], [[8.0]], [[4.0]]], device=image.device, dtype=image.dtype)  # 3x1x1
+    _HLS2RGB = tensor([[[0.0]], [[8.0]], [[4.0]]], device=image.device, dtype=image.dtype)  # 3x1x1
 
-    im: torch.Tensor = image.unsqueeze(-4)
-    h: torch.Tensor = torch.select(im, -3, 0)
-    l: torch.Tensor = torch.select(im, -3, 1)
-    s: torch.Tensor = torch.select(im, -3, 2)
+    im: Tensor = image.unsqueeze(-4)
+    h: Tensor = torch.select(im, -3, 0)
+    l: Tensor = torch.select(im, -3, 1)
+    s: Tensor = torch.select(im, -3, 2)
     h = h * (6 / math.pi)  # h * 360 / (2 * math.pi) / 30
     a = s * torch.min(l, 1.0 - l)
 
     # kr = (0 + h) % 12
     # kg = (8 + h) % 12
     # kb = (4 + h) % 12
-    k: torch.Tensor = (h + _HLS2RGB) % 12
+    k: Tensor = (h + _HLS2RGB) % 12
 
     # l - a * max(min(min(k - 3.0, 9.0 - k), 1), -1)
     mink = torch.min(k - 3.0, 9.0 - k)
     return torch.addcmul(l, a, mink.clamp_(min=-1.0, max=1.0), value=-1)
 
 
-# tricks to speed up a little bit the conversions by presetting small tensors
```

Comment:
I'm not sure of this

Reply:
Me neither,  it the only way to know it is measuring